pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarMetadataReader.java

```diff
@@ -245,7 +245,7 @@ public void commitOffsetToCursor(Map<TopicRange, MessageId> offset) {
             TopicRange tp = entry.getKey();
             try {
                 log.info("Committing offset {} to topic {}", entry.getValue(), tp);
-                admin.topics().resetCursor(tp.getTopic(), subscriptionNameFrom(tp), entry.getValue());
+                admin.topics().resetCursor(tp.getTopic(), subscriptionNameFrom(tp), entry.getValue(), true);
```

Comment:
Why change here? It causes unit tests to fail.

`TopicReaderTest#testCommitOffsetsToPulsar`

Reply:
The modification here is to make the submitted cursor meet the expectations of pulsar subscription, you can pay attention to issue #370

Reply:
Ok, Unit testing needs to be fixed. 

ExpectedId In `TopicReaderTest#testCommitOffsetsToPulsar`  test method  Should be changed to `Last message entryId + 1`

<img width="1529" alt="image" src="https://user-images.githubusercontent.com/33416836/124425940-e1080280-dd9b-11eb-910f-0f6bcd620b5b.png">

Reply:
I have a doubt, Since, We had reset cursor offset with exclusion. What's the meaning of param `excludeMessageId` in readerThread

Reply:
Should this be decided by the `StartupMode` or we always want to exclude the first message?

Reply:
@gaozhangmin @nlu90  The `excludeMessageId` is initially used to restore the offset from the flink state, because the location where the flink state is saved must be processed.

But it is also used in subscription mode activation. The cursor (referred to as NextMessageId) to obtain the first message to be read from the subscriber when the task is started. In the next, this cursor will be used to create a Reader consumer, and `excludeStartMessageId = true`. There are two situations:
1.  The subscriber's cursor is already the last message. The NextMessageId obtained must be greater than the latest message Id. Due to the check and processing of the cursor position, in this case, the latest message Id will be set to the startMessageId of the reader.
2.  There is data behind the subscriber's cursor. In this case, NextMessageId will be set to startMessageId.

**The startMessageId of Pulsar Reader achieved consistent performance in both cases.**